### PR TITLE
Makefile: simpily ledger build tags

### PR DESCRIPTION
**Pull request by MSevey**
In the process of working on the `goreleaser` PR I discovered that how we are building the `BUILD_TAGS` is overly verbose and unnecessary. 

Ultimately we are just trying to add `ledger` to `-tags`.  Hardcoding it and adding a note in the README for the 2 distributions that aren't supported seemed easier to maintain.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,52 +5,14 @@ DOCKER_BUF := $(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace bu
...
-# convert build_tags from a whitespace seperated list to a comma seperated list
-build_tags_comma_sep := $(subst $(whitespace),$(comma),$(build_tags))
-
 
 # process linker flags
 ldflags = -X github.com/cosmos/cosmos-sdk/version.Name=celestia-app \
 		  -X github.com/cosmos/cosmos-sdk/version.AppName=celestia-appd \
 		  -X github.com/cosmos/cosmos-sdk/version.Version=$(VERSION) \
 		  -X github.com/cosmos/cosmos-sdk/version.Commit=$(COMMIT) \
-		  -X "github.com/cosmos/cosmos-sdk/version.BuildTags=$(build_tags_comma_sep)"
```
Comment (MSevey) [inline Makefile:50]: Removing this line had zero impact on ledger it seemed. If someone knows why this line is needed, I'm happy to add it back in.

File: Makefile [hunk 2]
```diff
@@ -5,52 +5,14 @@ DOCKER_BUF := $(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace bu
...
-ldflags += $(LDFLAGS)
```
Comment (MSevey) [inline Makefile:51]: `LDFLAGS` isn't defined anywhere so this line is a no-op.

File: Makefile (hunk 1 above)
Comment (rootulp) [reply Makefile:50]: It appears to have been added in https://github.com/celestiaorg/celestia-app/pull/360 but `build_tags_comma_sep` wasn't defined in that PR so it seems like it was a no-op when first introduced.

**Review by rootulp — APPROVED** [review]
Thanks for the clean up!

[nit] PR title could use a conventional commit prefix so that the squashed commit has a conventional commit prefix so that the auto generated release notes are easy to scan and identify if the next release should be a major, minor, or patch release.

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**